pkg/controllers/leaderworkerset_controller.go

```diff
@@ -523,8 +544,54 @@ func (r *LeaderWorkerSetReconciler) iterateReplicas(ctx context.Context, lws *le
 	return continuousReadyReplicas, lwsUnreadyReplicas, nil
 }
 
+func (r *LeaderWorkerSetReconciler) getLeaderStatefulSet(ctx context.Context, lws *leaderworkerset.LeaderWorkerSet) (*appsv1.StatefulSet, error) {
+	sts := &appsv1.StatefulSet{}
+	err := r.Get(ctx, types.NamespacedName{Name: lws.Name, Namespace: lws.Namespace}, sts)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	return sts, nil
+}
+
+func (r *LeaderWorkerSetReconciler) createControllerRevisionIfNonExist(ctx context.Context, sts *appsv1.StatefulSet, lws *leaderworkerset.LeaderWorkerSet) (*appsv1.ControllerRevision, error) {
+	if sts == nil {
+		return revisionutils.CreateLeaderWorkerSetRevision(ctx, r.Client, lws, "")
+	}
+
+	stsRevision, err := revisionutils.GetLeaderWorkerSetRevisionFromTemplateHash(ctx, r.Client, lws, sts.Labels[leaderworkerset.TemplateRevisionHashKey])
+	if err != nil {
+		return nil, err
+	}
+
+	if stsRevision == nil {
+		return revisionutils.CreateLeaderWorkerSetRevision(ctx, r.Client, lws, sts.Labels[leaderworkerset.TemplateRevisionHashKey])
+	}
+
+	return revisionutils.CreateLeaderWorkerSetRevision(ctx, r.Client, lws, "")
```

Comment:
Since this is the only place where we create revisions, what happens when the LWS object changes? At this point, `leaderSts` exists, `stsRevision` exists, but we still need to create a new revision, which we don't here. it would return at line 566

Reply:
See this comment: https://github.com/kubernetes-sigs/lws/pull/277/files#r1898160442